### PR TITLE
Add libcurl to p2pool Dockerfile

### PR DESCRIPTION
**Pull request by bladedoyle**
Add libcurl to p2pool Dockerfile
Adjust a few monerod options

### CODE REVIEW

File: docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -25,19 +25,19 @@ services:
       - 18080:18080/tcp
     volumes:
       - monero:/home/monero/.bitmonero:rw
-      - /dev/null:/home/monero/.bitmonero/bitmonero.log:rw
       - /dev/hugepages:/dev/hugepages:rw
     restart: unless-stopped
     command: >-
       --zmq-pub tcp://0.0.0.0:18083
       --disable-dns-checkpoints
-      --enable-dns-blocklist
```
Comment (SChernykh) [inline docker-compose/docker-compose.yml:34]: Any specific reason for removing it?

Comment (jameswillhoite) [reply docker-compose/docker-compose.yml:34]: I would leave it. I had to add an additional list of blocked IP addresses as my monerod fell behind 12 blocks but it thought it was still 100% on main net.

Comment (bladedoyle) [reply docker-compose/docker-compose.yml:34]: Monero documentation says it is "Legacy - These options should no longer be necessary. They are still present in monerod for backwards compatibility."   

https://monerodocs.org/interacting/monerod-reference/#legacy

I have no problem leaving it if thats the consensus.  Please advise.

Comment (SChernykh) [reply docker-compose/docker-compose.yml:34]: I'm not sure why it's "legacy" and who categorized it. `--zmq-pub` is also under legacy section there and it's definitely not.

Comment (bladedoyle) [reply docker-compose/docker-compose.yml:34]: Ok.  I'll re-add it and push soon.  I always struggle with git command to amend or squish or whatever.  Wish me luck....or suggest the command if you have time.

Comment (bladedoyle) [reply docker-compose/docker-compose.yml]: I googled it :)
amend is easy, its the squish that confuses me.

**Comment by bladedoyle** [comment]
@SChernykh   Was there any change made for p2pool to exit on EOF or not having any STDIN?   The latest version exits immediately when run in a docker container (no terminal):

```
p2pool-p2pool  | 2022-06-15 00:12:24.0040 SideChain consensus ID = 22af7ee7************************************************4407f918
p2pool-p2pool  | 2022-06-15 00:12:24.0042 RandomX_Hasher allocated 2592 MB
p2pool-p2pool  | 2022-06-15 00:12:24.0045 ConsoleCommands started
p2pool-p2pool  | 2022-06-15 00:12:24.0065 ConsoleCommands EOF, stopping
p2pool-p2pool  | 2022-06-15 00:12:24.0065 P2Pool uv_run exited, result = 1
p2pool-p2pool  | 2022-06-15 00:12:24.0065 P2Pool stopped
```

monerod for example has --non-interactive flag to allow for this.

**Comment by SChernykh** [comment]
> Was there any change made for p2pool to exit on EOF or not having any STDIN? The latest version exits immediately when run in a docker container (no terminal):

Yes. It was probably some debugging that I did while adding libcurl, it shouldn't be there. Submitted the fix: https://github.com/SChernykh/p2pool/commit/921cbb0b69b81478b4aa3e490a409505467c2713

**Comment by bladedoyle** [comment]
Awesome.  That worked.
This PR is ready to merge.

However, there is another issue I see with p2pool where if monero rpc isnt up yet, the first call from p2pool fails and hangs.  No further attempts are made.  starting / restarting p2pool only after the moneord process is fully up does work, so I can update the container logic to wait ... but I think it would be much better if p2pool would continue to try even after an rpc call failure.  (note this is without using the rpc login and password flag)

**Review by SChernykh — APPROVED** [review]

**Comment by SChernykh** [comment]
> However, there is another issue I see with p2pool where if monero rpc isnt up yet, the first call from p2pool fails and hangs. No further attempts are made. 

It should keep retrying. I'll check it.

**Comment by SChernykh** [comment]
> if monero rpc isnt up yet, the first call from p2pool fails and hangs

I couldn't reproduce it during normal operation (not in docker), and it takes too much time to test in docker (monerod sync is slow). Can you start p2pool with log level 6 and paste the p2pool log?

**Pull request merged**